main.c

```diff
@@ -395,7 +397,8 @@ static void *dofastwork(void *task)
 			onionready(sname,secret,pubonion.raw);
 			pk[PUBLIC_LEN] = 0;
 			// don't reuse same seed
-			goto initseed;
+			if (!use_globalseed)
```

Comment:
this isn't that simple.
`addsztoscalar32(sk,counter);` above adds counter to secret key, which means that second time we find match we would add even bigger counter to already incrememented secret key, which would result in secret and public key mismatch.

Reply:
Ah, wasn't paying attention much how you keep the scalar with point in sync. Wouldn't just:

```
                        addsztoscalar32(sk,counter);
                        counter = 0;
```
do?

Reply:
There's a bigger issue with stopping that reseed altogether, the globalseed is bumped only at thread start, and (almost) never reseeded ever after inside the thread. User wouldn't be able to find same onions unless they run at least number of threads to reach the thread their original onion came from.

In addition to counting the delta properly, we need a stable point of reseed with trip count limit. Just reducing SIZE_MAX to something which can bump fairly regularly is enough. As a bonus, far more randomness can be soaked up for the search in non-passphrase setting as the reseed is now regular, not once SIZE_MAX / filter match.

Reply:
>User wouldn't be able to find same onions unless they run at least number of threads to reach the thread their original onion came from.

Yeah, that's one more backfire case.
But it's bad idea to penalize non-password cases because password use case isn't main purpose of this tool.
Sourcing randomness and SHA512 are both relatively more expensive than current way of adding stuff.
And current way of adding stuff don't make randomness any worse imo because there's enough space in key to accomodate (only 64 bits are possibly used of 256 bits).

Reply:
Again, you're talking overhead which is in not measurable. You can measure it only if you make trip count of 1000, but definitely not at 2^24. ge_add is so extremely slow that it will take *dozens of seconds* for the loop to run out. And then you spend 300ns grabbing random bits and bumping counter.

Now we can talk optimizations which *do* make sense. For instance, transforming prefixes into a mask for pre-compression field element so as to avoid p1p1_to_p3 and ge_p3_tobytes, which are after ge_add the second largest hogs in the profile.

Reply:
>User wouldn't be able to find same onions unless they run at least number of threads to reach the thread their original onion came from.

This is hella annoying...
Give me more time to think what to do.

Reply:
>For instance, transforming prefixes into a mask for pre-compression field element so as to avoid p1p1_to_p3 and ge_p3_tobytes, which are after ge_add the second largest hogs in the profile.

Yeah that would be nice to have. That would need a little bit deeper understanding of internals involved than I currently have, though. To be honest I don't really know how underlying field element primitives work, I just wanted tool to make vanity onions fast so I built it.

Reply:
Though wait, would that be even possible, isn't idea of these primitives that they can be multiplied fast, but division would be slow, and I see quite some multiplications both in `ge_add` and `ge_tobytes`..? Honest question, I have no idea.